esmvalcore/_recipe.py

```diff
@@ -293,6 +296,13 @@ def _get_default_settings(variable, config_user, derive=False):
             'end_year': variable['end_year'],
         }
 
+    if 'timerange' in variable and variable['frequency'] != 'fx':
+        settings['clip_start_end_year'] = {
+            'start_year': None,
```

Comment:
This is becoming a mess, changing the arguments of clip_start_end_year in 3 different places. Would it be possible to call the function `_update_timerange` just before `_get_default_settings` on line 830 and then set the values of the arguments just once to the final settings here?

Reply:
Sorry about the mess, but all those changes are not really needed for the functionality on it's own, they are just to avoid having backward incompatible changes. I'll try to get it to work more clearly.